test/test_videoapi.py

```diff
@@ -0,0 +1,213 @@
+import collections
+import os
+import unittest
+
+import torch
+import torchvision
+from torchvision.io import _HAS_VIDEO_OPT, VideoReader
+from torchvision.datasets.utils import download_url
+
+
+try:
+    import av
+
+    # Do a version test too
+    print("Success")
+    torchvision.io.video._check_av_available()
+except ImportError:
+    av = None
+
+
+VIDEO_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "assets", "videos")
+
+CheckerConfig = ["duration", "video_fps", "audio_sample_rate"]
+GroundTruth = collections.namedtuple("GroundTruth", " ".join(CheckerConfig))
+
+
+def fate(name, path="."):
+    """Download and return a path to a sample from the FFmpeg test suite.
+    See the `FFmpeg Automated Test Environment <https://www.ffmpeg.org/fate.html>`_
+    """
+
+    file_name = name.split("/")[1]
+    download_url("http://fate.ffmpeg.org/fate-suite/" + name, path, file_name)
+    return os.path.join(path, file_name)
+
+
+test_videos = {
+    "RATRACE_wave_f_nm_np1_fr_goo_37.avi": GroundTruth(
+        duration=2.0, video_fps=30.0, audio_sample_rate=None
+    ),
+    "SchoolRulesHowTheyHelpUs_wave_f_nm_np1_ba_med_0.avi": GroundTruth(
+        duration=2.0, video_fps=30.0, audio_sample_rate=None
+    ),
+    "TrumanShow_wave_f_nm_np1_fr_med_26.avi": GroundTruth(
+        duration=2.0, video_fps=30.0, audio_sample_rate=None
+    ),
+    "v_SoccerJuggling_g23_c01.avi": GroundTruth(
+        duration=8.0, video_fps=29.97, audio_sample_rate=None
+    ),
+    "v_SoccerJuggling_g24_c01.avi": GroundTruth(
+        duration=8.0, video_fps=29.97, audio_sample_rate=None
+    ),
+    "R6llTwEh07w.mp4": GroundTruth(
+        duration=10.0, video_fps=30.0, audio_sample_rate=44100
+    ),
+    "SOX5yA1l24A.mp4": GroundTruth(
+        duration=11.0, video_fps=29.97, audio_sample_rate=48000
+    ),
+    "WUzgd7C1pWA.mp4": GroundTruth(
+        duration=11.0, video_fps=29.97, audio_sample_rate=48000
+    ),
+}
+
+
+@unittest.skipIf(_HAS_VIDEO_OPT is False, "Didn't compile with ffmpeg")
+class TestVideoApi(unittest.TestCase):
+    def test_predefined_metadata(self):
+        """
+        Test that the source metadata corresponds to the one returned
+        by the new video decoder API.
+        """
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+            reader = VideoReader(full_path, "video")
+            reader_md = reader.get_metadata()
+            self.assertAlmostEqual(
+                config.video_fps, reader_md["video"]["fps"][0], delta=0.0001
+            )
+            self.assertAlmostEqual(
+                config.duration, reader_md["video"]["duration"][0], delta=0.5
+            )
+
+    def test_frame_reading(self):
```

Comment:
You need to skip this test if pyav is not available

Reply:
Done